matadi/_templates.py

```diff
@@ -137,6 +148,15 @@ def _fun_wrapper(self, x, **kwargs):
         F[2, 2] = 1  # fixed thickness ratio `h / H = 1`
         return self.fun(F, **kwargs)
 
+    def function(self, x, *args, **kwargs):
+        return self.W.function(x[:1], *args, **kwargs)
+
+    def gradient(self, x, *args, **kwargs):
+        return [*self.W.gradient(x[:1], *args, **kwargs), None]
+
+    def hessian(self, x, *args, **kwargs):
+        return self.W.hessian(x[:1], *args, **kwargs)
+
```

Comment:
This is the essential change: Take only the first item of `x` as state variables are unused within the hyperelastic materials.

**Hint**: No change is made to the attribute `.x` (does not include state variables).